packages/compass-indexes/src/components/create-index-modal/create-index-modal.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
+import {
+  ConfirmationModal,
+  Banner,
+  Disclaimer,
+  css,
+  spacing,
+} from '@mongodb-js/compass-components';
+
+import { toggleInProgress } from '../../modules/in-progress';
+import {
+  addField,
+  removeField,
+  updateFieldType,
+  updateFieldName,
+} from '../../modules/create-index/fields';
+import { nameChanged } from '../../modules/create-index/name';
+import { changeSchemaFields } from '../../modules/create-index/schema-fields';
+import {
+  createNewIndexField,
+  clearNewIndexField,
+} from '../../modules/create-index/new-index-field';
+import { clearError, handleError } from '../../modules/error';
+import { toggleIsVisible } from '../../modules/is-visible';
+import { toggleIsUnique } from '../../modules/create-index/is-unique';
+import { toggleUsePartialFilterExpression } from '../../modules/create-index/use-partial-filter-expression';
+import { toggleUseTtl } from '../../modules/create-index/use-ttl';
+import { ttlChanged } from '../../modules/create-index/ttl';
+import { toggleUseWildcardProjection } from '../../modules/create-index/use-wildcard-projection';
+import { toggleUseColumnstoreProjection } from '../../modules/create-index/use-columnstore-projection';
+import { wildcardProjectionChanged } from '../../modules/create-index/wildcard-projection';
+import { columnstoreProjectionChanged } from '../../modules/create-index/columnstore-projection';
+import { partialFilterExpressionChanged } from '../../modules/create-index/partial-filter-expression';
+import { toggleUseCustomCollation } from '../../modules/create-index/use-custom-collation';
+import { collationStringChanged } from '../../modules/create-index/collation-string';
+import { openLink } from '../../modules/link';
+import { createIndex } from '../../modules/create-index';
+import { resetForm } from '../../modules/reset-form';
+import type { CreateIndexProps } from '../create-index-form';
+import CreateIndexForm from '../create-index-form';
+import { toggleUseIndexName } from '../../modules/create-index/use-index-name';
+
+const { track } = createLoggerAndTelemetry('COMPASS-IMPORT-EXPORT-UI');
+
+const bannerStyles = css({
+  marginTop: spacing[3],
+});
+
+const createIndexModal = css({
```

Comment:
Should the modal have a minimum height or bottom padding before the buttons as well? On the default view opening the type select makes the height scroll:
<img width="448" alt="Screen Shot 2022-08-19 at 2 21 08 AM" src="https://user-images.githubusercontent.com/1791149/185556420-9e64e846-2d34-4d3c-8e20-ee91060a7b71.png">

Reply:
Replaced with Modal, but it didn't go away, so I set a min height for the modal to fix select.